src/rest/entries.ts

```diff
@@ -58,15 +24,23 @@ function updateNestedRef(
  * @param locale string - Locale code
  * @returns Entity - Updated REST response data
  */
-export function updateEntry(
+export function updateEntity(
   contentType: ContentType,
-  dataFromPreviewApp: Entity & { sys: SysProps },
-  updateFromEntryEditor: EntryProps,
+  dataFromPreviewApp: EntryProps,
+  updateFromEntryEditor: EntryProps | AssetProps,
   locale: string
-): Entity & { sys: SysProps } {
-  if (dataFromPreviewApp.sys.id === updateFromEntryEditor.sys.id) {
-    return mergeFields(contentType.fields, dataFromPreviewApp, updateFromEntryEditor, locale);
-  } else {
-    return updateNestedRef(contentType.fields, dataFromPreviewApp, updateFromEntryEditor, locale);
+): EntryProps {
+  for (const field of contentType.fields) {
+    const name = getFieldName(contentType, field);
+
+    if (isPrimitiveField(field) || field.type === 'RichText' || field.type === 'File') {
+      updatePrimitiveField(dataFromPreviewApp.fields, updateFromEntryEditor, name, locale);
+    } else if (field.type === 'Link') {
+      // update single ref field
```

Comment:
What's happening here? are these TODOs?

Reply:
yes it's this story: https://contentful.atlassian.net/browse/TOL-1025
(Was already this comment before, the position has only changed)